server/src/main/scala/controllers/Application.scala

```diff
@@ -139,10 +138,23 @@ class NoCacheFilter @Inject()(
   }
 }
 
-trait AvailableUserRoles {
+object UserRoleProvider {
```

Comment:
I'm thinking we should keep any decisions we make about using test code in a single place in Application.

ie, putting it where we currently have the DrtSystemInterface decision, rather than hiding it in this UserRoleProvider. We could have a TestUserRolesProvider and a ProdUserRolesProvider and decide which to use where we're also deciding which system interface to use.

Make sense?